File: algorithms/code/leetcode/lc191_number_of_1_bits/lc191_number_of_1_bits.py
#!/usr/bin/env python3
# https://leetcode.com/problems/number-of-1-bits/

import unittest


class Solution:
    def hammingWeight(self, n: int) -> int:
        count = 0
        for _ in range(0, 32):
            count += n % 2  # Faster than "& 1" in Python
            n >>= 1
        return count


class TestCode(unittest.TestCase):
    def test_11(self):
        self.assertEqual(3, Solution().hammingWeight(11))

File: algorithms/code/leetcode/lc191_number_of_1_bits/lc191_number_of_1_bits.rs
// https://leetcode.com/problems/number-of-1-bits/
// #easy

pub struct Solution;

impl Solution {
    pub fn hamming_weight(mut n: i32) -> i32 {
        let mut count = 0;
        for _ in 0..32 {
            count += n & 1;
            n >>= 1;
        }
        count
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_11() {
        assert_eq!(Solution::hamming_weight(11), 3);
    }
}
